### PR TITLE
[Jenkins] Add Solutions CFN Create VPC Test

### PR DESCRIPTION
**Pull request by lewijacn**
### Description
This change introduces a new Jenkins pipeline that synthesizes the Solutions Create VPC CFN template from this repository and deploys it to the test account. It will then perform the `initBootstrap` and ensure it exits properly as well as verifies that CLI tools such as cdk, docker, java, and python are now available on the box.

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-2228

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Jenkins AWS testing

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/migrationIntegPipelines/solutionsCFNTestCover.groovy
Comment (peternied) [inline jenkins/migrationIntegPipelines/solutionsCFNTestCover.groovy:2]: Does this need to be updated?

File: vars/solutionsCFNTest.groovy
Comment (peternied) [inline vars/solutionsCFNTest.groovy:37]: use `npm deploy` to reduce extra parameters like `--require-approval` and CDK doesn't need to be installed

File: test/awsRunInitBootstrap.sh
Comment (peternied) [inline test/awsRunInitBootstrap.sh:1]: I'd be hopeless at finding a bug in this test code, could we use python or typescript (via node)?

File: test/awsRunInitBootstrap.sh
Comment (peternied) [inline test/awsRunInitBootstrap.sh:104]: Verification seems pretty limited did you have plans to expand on this?

**Review by peternied — APPROVED** [review]
I really like that we can have some tests for this code, amazing! However, I'm a little concerned about building up shell script when we *could* lean on higher level languages, what do you think?

File: deployment/migration-assistant-solution/bin/app.ts
Comment (lewijacn) [reply deployment/migration-assistant-solution/bin/app.ts:42]: We can't set stack name once we are inside the stack correct?

File: jenkins/migrationIntegPipelines/solutionsCFNTestCover.groovy
Comment (lewijacn) [reply jenkins/migrationIntegPipelines/solutionsCFNTestCover.groovy:2]: Yes should be swapped back before merging, easier for testing right now

File: test/awsRunInitBootstrap.sh
Comment (lewijacn) [reply test/awsRunInitBootstrap.sh:1]: Yeah looking back now it may be beneficial to write this in python, I will create a task for this

File: test/awsRunInitBootstrap.sh
Comment (lewijacn) [reply test/awsRunInitBootstrap.sh:104]: Just wanted to check some basics right now since we have nothing, can definitely be enhanced in the future to check further

File: vars/solutionsCFNTest.groovy
Comment (lewijacn) [reply vars/solutionsCFNTest.groovy:37]: This only saves me not having to provide the `--require-approval` flag I believe and CDK CLI will already be on the box. My preference is to keep the same, but open if others want to change it.

Comment (peternied) [reply vars/solutionsCFNTest.groovy:37]: I'd really prefer we didn't have to run `npm install -g cdk` as part of the env setup, by using npm run it ensures everything is correctly version locked. 🧂

File: deployment/migration-assistant-solution/bin/app.ts
Comment (lewijacn) [reply deployment/migration-assistant-solution/bin/app.ts]: After talking with Peter, we can modify the id we pass into the constructor to have the suffix before making the `super()` call to achieve this

**Comment by lewijacn** [comment]
> I really like that we can have some tests for this code, amazing! However, I'm a little concerned about building up shell script when we _could_ lean on higher level languages, what do you think?

Create JIRA to handle this: https://opensearch.atlassian.net/browse/MIGRATIONS-2231

File: test/awsRunInitBootstrap.sh
Comment (lewijacn) [reply test/awsRunInitBootstrap.sh:1]: https://opensearch.atlassian.net/browse/MIGRATIONS-2231

**Pull request merged**
